src/nyx/features/image_matrix.cpp

```diff
@@ -120,6 +102,68 @@ void ImageMatrix::print (const std::string& head, const std::string& tail, std::
 	std::cout << tail;
 }
 
+void ImageMatrix::print (std::ofstream& f, const std::string& head, const std::string& tail, std::vector<PrintablePoint> special_points)

```

Comment:
Method print() is intended for printing diagnostic output following some error, never for efficient execution. The reason of passing the special points argument by value rather than by reference is to enable a default value "empty set" for this parameter as "std::vector<PrintablePoint> special_points = {}", see declaration of method print() in file image_matrix.h. 


calling .print() as 
someImageMatrix.print(ss.str(), "", { {pxO.x, pxO.y, "(O)"},  {pxA.x, pxA.y, "(A)"}, {pxContour.x, pxContour.y, "(C)"} })
and the reference declaration wouldn't let this method call to compile due to an anonymous vector instance {...}. Since this code